includes/class-swedbank-pay-scheduler.php

```diff
@@ -0,0 +1,144 @@
+<?php
+/**
+ * Swedbank Pay Scheduler Class.
+ * Handles scheduled tasks for Swedbank Pay.
+ *
+ * @package SwedbankPay\Checkout\WooCommerce
+ */
+
+namespace SwedbankPay\Checkout\WooCommerce;
+
+use WC_Logger;
+
+defined( 'ABSPATH' ) || exit;
+
+
+/**
+ * Swedbank_Pay_Scheduler Class.
+ */
+class Swedbank_Pay_Scheduler {
+	public const ACTION_ID = 'swedbank_pay_scheduler_run';
+
+	/**
+	 * @var Swedbank_Pay_Scheduler
+	 */
+	private static $instance = null;
+
+	/**
+	 * Singleton pattern
+	 *
+	 * @return Swedbank_Pay_Scheduler
+	 */
+	public static function get_instance() {
+		if ( null === self::$instance ) {
+			self::$instance = new self();
+		}
+
+		return self::$instance;
+	}
+
+	/**
+	 * Logger instance.
+	 *
+	 * @var WC_Logger
+	 */
+	private $logger;
+
+	/**
+	 * Class constructor.
+	 */
+	public function __construct() {
+		$this->logger = \wc_get_logger();
+
+		add_action( 'swedbank_pay_scheduler_run', array( $this, 'run' ), 10, 2 );
+	}
+
+	/**
+	 * Log message.
+	 *
+	 * @param string $message The message to log.
+	 */
+	private function log( $message ) {
+		$this->logger->info( $message, array( 'source' => 'swedbank_pay_queue' ) );
+	}
+
+	/**
+	 * Code to execute for each item in the queue.
+	 *
+	 * @throws \Exception If the webhook data is invalid or if the order cannot be found.
+	 *
+	 * @param string $payment_method_id The payment method ID.
+	 * @param string $webhook_data The webhook data in JSON format.
+	 *
+	 * @return false|null
+	 */
+	public function run( $payment_method_id, $webhook_data ) {
+		$this->log( sprintf( 'Start task: %s', wp_json_encode( array( $payment_method_id, $webhook_data ) ) ) );
+
+		try {
+			$data = json_decode( $webhook_data, true );
+			if ( empty( $data ) ) {
+				throw new \Exception( 'Invalid webhook data' );
```

Comment:
Suggestion (non-blocking): I would recommend using `WP_Exception` over just `Exception`. The generic `Exception` type should preferably never be used or caught unless you need to ensure that no error should stop the process ever.